demos/http/http_demo_s3_download_multithreaded/http_demo_s3_download_multithreaded.c

```diff
@@ -161,6 +161,24 @@ typedef struct ResponseItem
     uint8_t responseBuffer[ USER_BUFFER_LENGTH ];
 } ResponseItem_t;
 
+/* @brief Struct used for sending requests to the HTTP thread.
+ *
+ * This is used by both the main and HTTP threads, though since we create the
+ * HTTP thread without a shared memory space, each will have its own copy. We
+ * have this as a global so that it will be located at the same address in the
+ * main and HTTP threads so that way pointers remain valid when copied over.
```

Comment:
How do the main and HTTP threads synchronize?  What prevents one task from overwriting one of these values while the other one is reading them?  Is there any critical section protection?